src/NLog.Targets.Seq/LogEventInfoFormatter.cs

```diff
@@ -49,58 +51,87 @@ static LogEventInfoFormatter()
             };
         }
 
-        public static void ToCompactJson(IEnumerable<LogEventInfo> events, TextWriter output, IList<SeqPropertyItem> properties)
+        public static void ToCompactJson(IEnumerable<LogEventInfo> logEvents, TextWriter output, IList<SeqPropertyItem> properties)
         {
-            foreach (var loggingEvent in events)
+            foreach (var logEvent in logEvents)
             {
-                ToJson(loggingEvent, output, properties);
+                ToCompactJson(logEvent, output, properties);
                 output.Write(Environment.NewLine);
             }
         }
 
-        static void ToJson(LogEventInfo loggingEvent, TextWriter output, IList<SeqPropertyItem> properties)
+        public static void ToCompactJson(LogEventInfo logEvent, TextWriter output, IList<SeqPropertyItem> properties)
         {
-            if (loggingEvent == null) throw new ArgumentNullException(nameof(loggingEvent));
+            if (logEvent == null) throw new ArgumentNullException(nameof(logEvent));
             if (output == null) throw new ArgumentNullException(nameof(output));
             if (properties == null) throw new ArgumentNullException(nameof(properties));
 
+            // Without this call, logEventInfo.Properties behaves erratically when templates are used; stll trying
+            // to track down the cause (needs some debugging into NLog).
+            logEvent.FormattedMessage?.ToString();
```

Comment:
The problem appears to be that `LogEvent.CalcFormattedMessage()` (private) actually populates the `Properties` collection:

https://github.com/NLog/NLog/commit/67368ec3f0840cb7c1c312a51225c50ff7abb84a#diff-fe47bd426efb3db67d05e56b297587c6R509

Would it be less of a footgun if the first call to read `Properties` invoked this logic? Formatting wouldn't be required, but unless the template is fully processed, `Properties` won't be complete.